src/domain/enrolment/CreateEnrolmentPage.tsx

```diff
@@ -68,18 +69,27 @@ const CreateEnrolmentPage: React.FC = () => {
   const initialValues = React.useMemo(
     () => ({
       ...defaultInitialValues,
+      language: locale.toUpperCase(),
+      // some of the values used only for validation purposes
       minGroupSize: Math.max(
         ...filteredOccurrences.map((item) => item?.minGroupSize || 0)
       ),
+      // figure out maxGroup size based on all occurrences selected
+      // if OccurrenceType is EnrolmentCount, use only maxGroupSize
       maxGroupSize: Math.min(
-        ...filteredOccurrences.map((item) =>
-          Math.min(
-            item?.maxGroupSize || item.amountOfSeats,
-            item.amountOfSeats - (item.seatsTaken || 0)
-          )
-        )
+        ...filteredOccurrences.map((item) => {
+          if (item.seatType === OccurrenceSeatType.ChildrenCount) {
+            return Math.min(
+              item?.maxGroupSize || item.amountOfSeats,
+              item.amountOfSeats - (item.seatsTaken || 0)
+            );
+          }
+          if (item.seatType === OccurrenceSeatType.EnrolmentCount) {
```

Comment:
A trivial thing, but this if-statement could be else-if.

Reply:
I thought that when writing this code but it is 50/50 which style I want to use. To me in this case where we only return something inside the if, else keywords just adds noise.

I prefer this "early return" style.

Reply:
Tht being said it is not 50/50 for me :D

Reply:
Did some googling.. some disuccions about that here:

https://dev.to/tyrw/ifelse-or-just-if-1kie

Reply:
One more option is switch + assertUnreachable code check: https://stackoverflow.com/questions/39419170/how-do-i-check-that-a-switch-block-is-exhaustive-in-typescript
Good thing is that if we add new enum type to OccurrenceSeatType then typescript will complain if new type ismissing on switch-case.

Reply:
assertUnreachable style is very nice pattern

Reply:
```
        ...filteredOccurrences.map((item) => {
          if (item.seatType === OccurrenceSeatType.ChildrenCount) {
            return Math.min(
              item?.maxGroupSize || item.amountOfSeats,
              item.amountOfSeats - (item.seatsTaken || 0)
            );
          }
          if (item.seatType === OccurrenceSeatType.EnrolmentCount) {
            return item?.maxGroupSize || 0;
          }
          return assertUnreachable(item.seatType);
        })
```

That works also

Reply:
<img width="600" alt="Screenshot 2021-02-10 at 8 49 17" src="https://user-images.githubusercontent.com/15219142/107475876-82888080-6b7d-11eb-9217-d11548768d9e.png">
Decided to use switch case because it liiks the cleanest to me :) But left that assertion to default case because otherwise ts complains

Reply:
Looks nice!